### ISSUE TITLE
runtime_error during NEDeconvolutionLayer configure

### ISSUE THREAD

**Issue by pranathibl** [ARM-software/ComputeLibrary#883]
Hi,

I am getting the following error with NEDeconvolutionLayer with v20.11.

terminate called after throwing an instance of 'std::runtime_error'
  what():  in validate_and_configure_window src/core/NEON/kernels/NEDirectConvolutionLayerKernel.cpp:944: Not implemented
Aborted

The same setup was working fine with v20.02.1. but with 20.11 I am facing this issue. 

setup to reproduce this:
[transposed_standalone.zip](https://github.com/ARM-software/ComputeLibrary/files/6146795/transposed_standalone.zip)

Thanks
Pranathi

**Comment by pranathibl** [comment]
Hi @morgolock 

Any update on this? 
Are there any specific configurations of inputs for which this will fail?

-Pranathi

**Comment by morgolock** [comment]
Hi @pranathibl 

In general, it's a good practice to call the function's validate() method to check if the configuration is supported.

In this particular case, validate fails because the configuration layout NCHW and kernel size 16 is not supported by NEDirectConvolution. NEDeconvolutionLayer uses NEConvolutionLayer to compute the result.
 
Hope this helps.

**Comment by pranathibl** [comment]
Hi @morgolock ,

The same setup was working with v20.02.1 and not with v20.11. Is implementation of Deconv layer changed?

Thanks
Pranathi

**Comment by morgolock** [comment]
Hi @pranathibl 

Yes, DeconvolutionLayer uses ConvolutionLayer and the heuristic to choose the convolution method has changed since 20.02.

https://github.com/ARM-software/ComputeLibrary/blob/master/src/runtime/NEON/functions/NEConvolutionLayer.cpp#L58

In 20.02 Convolution method for this workload was GEMM whereas in 20.11 is DirectConv.

Hope this helps.

**Comment by pranathibl** [comment]
Hi @morgolock 

I understand that convolution method picked up now is different from v20.02.1. But how can i support DeConv layer with such configuration?
Is there any way to support it, As it is supported in previous releases?

-Pranathi

**Comment by morgolock** [comment]
Hi @pranathibl 

Please see below a work-around, please beware that this disables the DIrectConv method and will have side effects on other workloads.

At present the library does not provide a way to override the convolution method but we are going to analyse if this is something that could be beneficial. 

As explained above, the heuristic to choose the convolution method has changed and the new method chosen by the library does not support your workload kernel size 16

```
diff --git a/src/runtime/NEON/functions/NEConvolutionLayer.cpp b/src/runtime/NEON/functions/NEConvolutionLayer.cpp
index e43d976944..4ee9fb3784 100644
--- a/src/runtime/NEON/functions/NEConvolutionLayer.cpp
+++ b/src/runtime/NEON/functions/NEConvolutionLayer.cpp
@@ -180,13 +180,6 @@ ConvolutionMethod NEConvolutionLayer::get_convolution_method(const ITensorInfo *
     }
     else
     {
-        // SRGAN
-        // Output might not be initialized when it is an internal tensor of the layer using the convolution
-        if(input->total_size() > 1e7 && (weights->dimension(idx_h) > 7)
-           && (NEDirectConvolutionLayer::validate(input, weights, nullptr, output, conv_info, act_info)))
-        {
-            return ConvolutionMethod::DIRECT;
-        }
         if((weights->dimension(idx_h) > 7) && (input->dimension(idx_c) > output->dimension(idx_c)) && (NEFFTConvolutionLayer::validate(input, weights, nullptr, output, conv_info, act_info)))
         {
             return ConvolutionMethod::FFT;
```

Hope this helps.

**Comment by morgolock** [comment]
Hi @pranathibl 

The patch below enables the old GEMM code path for NCHW layout.

https://review.mlplatform.org/c/ml/ComputeLibrary/+/5935

This will be included in the next release v21.08.

Please reopen if you still need support.
